### ISSUE TITLE
node.run_state['root_path'] conflicts with pyenv cookbook

### ISSUE THREAD

**Issue by petracvv** [sous-chefs/ruby_rbenv#285]
## :ghost: Brief Description

When using both the rbenv and pyenv cookbooks together and installing python and ruby versions for the same user, the resources from the 2nd cookbook do not work. This is because the same internal run_state attribute is used for tracking the root_path in both cookbooks.

## :pancakes: Cookbook version

2.4.0

## :woman_cook:  Chef-Infra Version

14.13.11

## :tophat: Platform details

RHEL7

## Steps To Reproduce

1. Write cookbook that creates a user and installs a ruby and python version for that user:
  ```ruby
  user 'jenkins' do
    manage_home true
  end

  rbenv_user_install 'jenkins'

  rbenv_ruby '2.7.1' do
    user 'jenkins'
  end

  pyenv_user_install 'jenkins'

  pyenv_python '3.8.2' do
    user 'jenkins'
  end
```
2. Run the recipe against a node in test-kitchen or a test env.
3. See the pyenv python installation fail

## :police_car: Expected behavior

I expect that both the ruby and python versions are installed with no error.

## :heavy_plus_sign: Additional context

I can work around the problem with a ruby block in between the pyenv and rbenv resources:
```ruby
ruby_block 'unset run_state' do
  block do
    node.run_state['root_path'] = {}
  end
end
```

**Comment by damacus** [comment]
Both cookbooks need to be namespaced really. Mind raising an issue over there too please?

**Comment by petracvv** [comment]
Yeah I can make another issue there.

**Comment by xorima** [comment]
See comment's in pyenv https://github.com/sous-chefs/pyenv/issues/68

Basically we need to namespace this

**Comment by damacus** [comment]
Fixed via #287
